src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -805,4 +809,56 @@ export class PatientViewPageStore {
         this.ajaxErrors = [];
     }
 
+    readonly trialMatches = remoteData<ITrialMatch[]>({
+        invoke: () => {
+            return postMatchMinerTrialMatches({mrn: this.patientId});
+        },
+        onError: (err: Error) => {
+            // fail silently
+        }
+    }, []);
+
+    readonly trialIds = remoteData<ITrialQuery>({
+        await: () => [
+            this.trialMatches
+        ],
+        invoke: async() => {
+            let nctIds = new Set<string>();
+            let protocolNos = new Set<string>();
+            _.forEach(this.trialMatches.result, (trialMatch: ITrialMatch) => {
+                if (_.isEmpty(trialMatch.protocolNo)) {
+                    nctIds.add(trialMatch.nctId);
+                } else {
+                    protocolNos.add(trialMatch.protocolNo);
+                }
+            });
+            return {
+                nct_id: [...nctIds],
+                protocol_no: [...protocolNos]
+            };
+        },
+        onError: (err: Error) => {
+            // fail silently
+        }
+    }, {
+        nct_id: [],
+        protocol_no: []
+    });
+
+    readonly matchMinerTrials = remoteData<ITrial[]>({
+        await: () => [
+            this.trialIds
+        ],
+        invoke: async () => {
+            if (this.trialIds.result.protocol_no.length > 0 || this.trialIds.result.nct_id.length > 0) {
+                return postMatchMinerTrialsById(this.trialIds.result);
```

Comment:
this isn't a good name for what I think is a fetch function (gets trials).  Follow naming scheme of existing clients.   fetchSomethingUsingPost

Reply:
changed